log4j-core/src/main/java/org/apache/logging/log4j/core/util/internal/instant/InstantPatternDynamicFormatter.java

```diff
@@ -47,6 +47,30 @@
  * <li>Precompute and cache the output for parts that are of precision lower than or equal to {@value InstantPatternDynamicFormatter#PRECISION_THRESHOLD} (i.e., {@code yyyy-MM-dd'T'HH:mm:} and {@code X}) and cache it</li>
  * <li>Upon a formatting request, combine the cached outputs with the dynamic parts (i.e., {@code ss.SSS})</li>
  * </ol>
+ * <h2>Implementation note</h2>
+ * <p>
+ * Formatting can actually even be made faster and garbage-free by manually formatting sub-minute precision directives as follows:
```

Comment:
I guess that the `DateTimeFormatter` does not become garbage-free on its own, when it is hot enough?

Reply:
@ppkarwasz, I indeed doubt that. If you'd skim through `DateTimeFormatter#formatTo()`, there are several new object instantiations.